api/response/schema.py

```diff
@@ -69,21 +94,93 @@ class Query(graphene.ObjectType):
     """
         Query to get the room response
     """
-    get_room_response = graphene.List(
-        Response,
+    get_room_response = graphene.Field(
+        PaginatedResponse,
         room_id=graphene.Int(),
+        page=graphene.Int(),
+        per_page=graphene.Int(),
         description="Returns a list of responses of a room. Accepts the arguments\
-            \n- room_id: Unique identifier of a room"
-    )
+            \n- room_id: Unique identifier of a room\
+            \n- page: Page number of responses\
+            \n- per_page: Number of room responses per page")
+
+    def map_room_responses(self, responses):
+        mapped_response = []
+        missing_resource = []
+        for response in responses:
+            response_id = response.id
+            suggestion = response.text_area
+            created_date = response.created_date
+            rating = response.rate
+            resolved = response.resolved
+            if len(response.missing_resources) > 0:
+                for resources in response.missing_resources:
+                    resource_name = resources.name
+                    missing_resource.append(resource_name)
+                response_in_room = ResponseDetail(
+                    response_id=response_id,
+                    suggestion=suggestion,
+                    created_date=created_date,
+                    rating=rating,
+                    missing_items=missing_resource,
+                    resolved=resolved)
+                missing_resource = []
```

Comment:
missing_resource is initialized in line 108 as an empty list, why is it being declared here again?

Reply:
That must've been a mistake. I'll fix.